CMakeLists.txt

```diff
@@ -82,8 +89,10 @@ message(STATUS "DEBUG_SYMBOL: ${DEBUG_SYMBOL}")
 if(DEBUG_SYMBOL)
     if(MSVC)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /DEBUG")
+        add_compile_options(/O2 /Ob2)
```

Comment:
unfortunately, pybind11 has bug for -O0, customized caster will not work.
explicitly set -O2 for debug build.

Reply:
What is the pybind11 bug with `O0` and debug build?

Setting `O2` may be a workaround but it may not be merged without describing the root cause clearly.  A clear description must be given when the issue is found, or it will be overly difficult for reproduction later.